app/src/main/java/com/example/cinemapp/ui/splash/SplashActivity.kt

```diff
@@ -5,17 +5,19 @@ import android.os.Bundle
 import androidx.activity.enableEdgeToEdge
 import androidx.appcompat.app.AppCompatActivity
 import androidx.lifecycle.lifecycleScope
+import com.example.cinemapp.data.UserPreferences
 import com.example.cinemapp.databinding.ActivitySplashBinding
+import com.example.cinemapp.ui.authentication.AuthenticationActivity
 import com.example.cinemapp.ui.main.MainActivity
-import kotlinx.coroutines.Dispatchers
-import kotlinx.coroutines.delay
 import kotlinx.coroutines.launch
+import org.koin.android.ext.android.inject
 import org.koin.androidx.viewmodel.ext.android.viewModel
 
 class SplashActivity : AppCompatActivity() {
 
     private val viewModel by viewModel<SplashViewModel>()
     private lateinit var binding: ActivitySplashBinding
+    private val userPrefs: UserPreferences by inject()
```

Comment:
Again, `UserPreferences` should not be mentioned in `SplashActivity`, but in its ViewModel